docs/scos/dev/feature-integration-guides/202200.0/glue-api/glue-json-api-convention-integration-guide.md

```diff
@@ -0,0 +1,502 @@
+
+        return $routeCollection;
+    }
+}
+```
+
+**src/Pyz/Glue/DummyModuleApi/Plugin/DummyResourceRelationshipPlugin.php**
+```php
+<?php
+
+namespace Spryker\Glue\DummyModuleApi\Plugin;
+
+use Generated\Shared\Transfer\GlueRelationshipTransfer;
+use Generated\Shared\Transfer\GlueRequestTransfer;
+use Generated\Shared\Transfer\GlueResourceTransfer;
+use Spryker\Glue\GlueJsonApiConventionExtension\Dependency\Plugin\ResourceRelationshipPluginInterface;
+use Spryker\Glue\Kernel\AbstractPlugin;
+
+class DummyCountriesResourceRelationshipPlugin extends AbstractPlugin implements ResourceRelationshipPluginInterface
+{
+    /**
+     * @var string
+     */
+    protected const RESOURCE_TYPE_DUMMY_RELATIONSHIPS = 'dummyRelationshipsType';
+
+    /**
+     * @api
+     *
+     * @param array<\Generated\Shared\Transfer\GlueResourceTransfer> $resources
+     * @param \Generated\Shared\Transfer\GlueRequestTransfer $glueRequestTransfer
+     *
+     * @return void
+     */
+    public function addRelationships(array $resources, GlueRequestTransfer $glueRequestTransfer): void
+    {
+        foreach ($resources as $glueResourceTransfer) {
+            $glueRelationshipTransfer = (new GlueRelationshipTransfer())
+                ->addResource(new GlueResourceTransfer());
+            $glueResourceTransfer->addRelationship($glueRelationshipTransfer);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @api
+     *
+     * @return string
+     */
+    public function getRelationshipResourceType(): string
+    {
+        return static::RESOURCE_TYPE_DUMMY_RELATIONSHIPS;
+    }
+}
+```
+
+### 5) Set up behavior
+Enable the following behaviors by registering the plugins:
+
+| PLUGIN                                                | SPECIFICATION                                                                   | NAMESPACE                                                                                                   |
+|-------------------------------------------------------|---------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------|
+| HttpCommunicationProtocolPlugin                       | Defines applicability, accepts the request and sends the response.              | Spryker\Glue\GlueHttp\Plugin\GlueApplication                                                                |
+| CorsHeaderExistenceRequestAfterRoutingValidatorPlugin | Executes validations that need to be aware of the resolved route.               | Spryker\Glue\GlueHttp\Plugin\GlueStorefrontApiApplication                                                   |
+| ConventionResourceFilterPlugin                        | Filters resources for JSON API convention.                                      | Spryker\Glue\GlueApplication\Plugin\GlueApplication                                                         |
```

Comment:
Filters out Resources that do not implement the convention current request is using.